src/main/kotlin/net/lumalyte/lg/infrastructure/services/GuildServiceBukkit.kt

```
@@ -474,6 +474,49 @@ class GuildServiceBukkit(
+            if (!hasPermission(actorId, guildId, RankPermission.MANAGE_MEMBERS)) {
+                logger.warn("Player $actorId attempted to remove ally home for guild $guildId without permission")
+                return false
+            }
+            if (guild.allyHome == null) return false
+            val updatedGuild = guild.copy(allyHome = null)
+            val result = guildRepository.update(updatedGuild)
+            if (result) {
+                logger.info("Guild $guildId ally home removed by $actorId")
+            }
+            return result
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)